tests/proofs/memory-concrete-type-spec.k

```diff
@@ -0,0 +1,43 @@
+requires "kwasm-lemmas.k"
+
+module MEMORY-CONCRETE-TYPE-SPEC
+    imports WASM-TEXT
+    imports KWASM-LEMMAS
+
+    rule <k> (i32.store (i32.const ADDR) (i32.load (i32.const ADDR)):Instr):Instr => . ... </k>
+         <curModIdx> CUR </curModIdx>
+         <moduleInst>
+           <modIdx> CUR </modIdx>
+           <memAddrs> 0 |-> MEMADDR </memAddrs>
+           ...
+         </moduleInst>
+         <memInst>
+           <mAddr> MEMADDR </mAddr>
+           <msize> SIZE </msize>
+           <mdata> BM   </mdata>
+           ...
+         </memInst>
+       requires #chop(<i32> ADDR) ==K <i32> EA
+        andBool EA +Int #numBytes(i32) <=Int SIZE *Int #pageSize()
+        andBool #isByteMap(BM)
+        andBool #inUnsignedRange(i32, ADDR)
+
+    rule <k> (i64.store (i32.const ADDR) (i64.load (i32.const ADDR)):Instr):Instr => . ... </k>
+         <curModIdx> CUR </curModIdx>
+         <moduleInst>
+           <modIdx> CUR </modIdx>
+           <memAddrs> 0 |-> MEMADDR </memAddrs>
+           ...
+         </moduleInst>
+         <memInst>
+           <mAddr> MEMADDR </mAddr>
+           <msize> SIZE </msize>
+           <mdata> BM   </mdata>
+           ...
+         </memInst>
+       requires #chop(<i32> ADDR) ==K <i32> EA
+        andBool EA +Int #numBytes(i64) <=Int SIZE *Int #pageSize()
+        andBool #isByteMap(BM)
+        andBool #inUnsignedRange(i32, ADDR)
+
+endmodule
```

Comment:
Isn't this just an instance of the abstract type proof? If so, are we having this one so it can run with the Haskell backend? In that case, let's try a submodule bump and see if it works with latest haskell backend. If not, let's notify the haskell backend team.

Reply:
I wanted the `kwasm-lemmas` to do more granular memory reasoning. With this setup, the above proof goes through without the `#setRange(#getRange)` lemma, just using modular arithmetic. This works becuse when the type is concrete, the `#getRange` and `#setRange` functions get expanded into a series of `#get` and `#set`, and the new lemmas can do reasoning over those.

Reply:
I'll do the submodule bump and see if it works.

Reply:
So to exercise some more z3 level reasoning about memory I switched to `store16` and `load32_u` to make sure the `#setRange(#getRange)` lemma doesn't apply.